app/_components/SubmitGuessButton.tsx

```diff
@@ -110,32 +68,29 @@ function SubmitGuessButton({
           body: JSON.stringify(data),
         }
       );
-      // checking the tracking pin
-      for (const pin of pins) {
-        if (pin.poi_id === trackingPin?.poi_id) {
-          trackingPin.is_completed = true;
-        }
-      }
-      console.log("response", response);
+      const { message } = await response.json() as {message: string};
+      console.log(message);
     } catch (error) {
-      console.error("Error", error);
+      console.error(error);
```

Comment:
The `throw` error above is a good idea, but this `catch` block isn't doing anything with it besides logging it in the console.

It might be good to return some kind of response (e.g., an "errors" object) with descriptive errors to be displayed in the form.